search/includes/classes/class-search.php

```diff
@@ -128,6 +130,43 @@ public function action__plugins_loaded() {
 		}
 	}
 
+	public function action__wp() {
+		global $wp_query;
+
+		// Avoid infinite loops because our requests load the URL with this param.
+		if ( isset( $_GET['es'] ) ) {
+			return;
+		}
+
+		// Temp functionality for testing phase.
+		// If this was a regular search page and VIP Search was _not_ used, and if the site is configured to do so,
+		// re-run the same query, but with `es=true`, via JS to test both systems in parallel
+		if ( is_search() && ! isset( $wp_query->elasticsearch_success ) ) {
+			$is_enabled_by_constant = defined( 'VIP_ENABLE_SEARCH_QUERY_MIRRORING' ) && true === VIP_ENABLE_SEARCH_QUERY_MIRRORING;
+
+			$option_value = get_option( 'vip_enable_search_query_mirroring' );
+			$is_enabled_by_option = in_array( $option_value, array( true, 'true', 'yes', 1, '1' ), true );
+
+			$is_mirroring_enabled = $is_enabled_by_constant || $is_enabled_by_option;
+
+			if ( $is_mirroring_enabled ) {
+				add_action( 'shutdown', [ $this, 'do_mirror_search_request' ] );
+			}
+		}
+	}
+
+	public function do_mirror_search_request() {
+		fastcgi_finish_request();
+
+		$vip_search_url = home_url( add_query_arg( 'es', 'true' ) );
+
+		wp_remote_request( $vip_search_url, [
```

Comment:
Open question here about whether this non-blocking request is good enough or whether we should use the `vip_safe_` function.